src/ts/ts-genapi/export.ts

```diff
@@ -0,0 +1,356 @@
+            {
+                this.newline();
+            }
+        });
+    }
+}
+
+declare const enum ApiViewTokenKind {
+    Text = 0,
+    Newline = 1,
+    Whitespace = 2,
+    Punctuation = 3,
+    Keyword = 4,
+    LineIdMarker = 5, // use this if there are no visible tokens with ID on the line but you still want to be able to leave a comment for it
+    TypeName = 6,
+    MemberName = 7,
+    StringLiteral = 8
+}
+
+declare interface IApiViewFile {
+    Name: string;
+    Tokens: IApiViewToken[];
+    Navigation: IApiViewNavItem[];
+}
+
+declare interface IApiViewToken {
+    Kind: ApiViewTokenKind;
+    DefinitionId?: string;
+    NavigateToId?: string;
+    Value?: string;
+}
+
+declare interface IApiViewNavItem {
+    Text: string;
+    NavigationId: string;
+    ChildItems: IApiViewNavItem[];
+    Tags: {
+        [propertyName: string]: string;
+    };
+}
+
+function appendMembers(builder: TokensBuilder, navigation: IApiViewNavItem[], item: ApiItem)
+{
+    builder.lineId(item.canonicalReference.toString());
+    builder.indent();
+    if (item instanceof ApiDeclaredItem) {
+        item.excerptTokens.forEach(token => {
+            if (token.kind == ExcerptTokenKind.Reference)
+            {
+                builder.typeReference(token.canonicalReference.toString(), token.text);
+            }
+            else
+            {
+                builder.splitAppend(token.text, item.canonicalReference.toString(), item.displayName);
+            }
+        });
+    }
+
+    var navigationItem: IApiViewNavItem;
+    var typeKind: string;
+
+    switch (item.kind)
+    {
+        case ApiItemKind.Interface:
+        case ApiItemKind.Class:
+            typeKind = item.kind.toLowerCase();
+            break
+        case ApiItemKind.TypeAlias:
+            typeKind = "struct";
+            break
+    }
+
+    if (typeKind != null)
+    {
+        navigationItem = {
+            Text: item.displayName,
+            NavigationId: item.canonicalReference.toString(),
+            Tags: {
+                "TypeKind": typeKind
+            },
+            ChildItems: []
+        };
+        navigation.push(navigationItem);
+    }
+
+    if (item.kind == ApiItemKind.Interface ||
+        item.kind == ApiItemKind.Class)
+    {
+        if (item.members.length > 0)
+        {
+            builder
+                .punct("{")
+                .newline()
+                .incIndent()
+    
+            item.members.forEach(member => {
+                appendMembers(builder,navigationItem.ChildItems, member);
+            });
+
+            builder
+                .decIndent()
+                .indent()
+                .punct("}")
+                .newline();
+        }
+        else
+        {
+            builder
+                .punct("{")
+                .space()
+                .punct("}")
+                .newline();
+        }
+    }
+    else
+    { 
+        builder.newline();
+    }
+}
+
+const apiModel: ApiModel = new ApiModel();
```

Comment:
```suggestion
const apiModel = new ApiModel();
```

no need to annotate types when they're directly inferable